nexus/src/db/model.rs

```diff
@@ -352,6 +353,55 @@ pub struct Ipv6Net(pub external::Ipv6Net);
 NewtypeFrom! { () pub struct Ipv6Net(external::Ipv6Net); }
 NewtypeDeref! { () pub struct Ipv6Net(external::Ipv6Net); }
 
+impl Ipv6Net {
+    /// Generate a random subnetwork from this one, of the given prefix length.
+    ///
+    /// `None` is returned if:
+    ///
+    ///  - `prefix` is less than this address's prefix
+    ///  - `prefix` is greater than 128
+    pub fn random_subnet(&self, prefix: u8) -> Option<Self> {
+        use rand::RngCore;
```

Comment:
I did indeed [add a few tests](https://github.com/oxidecomputer/omicron/blob/0299cbfc13d77662e752848e5a8715505986602f/nexus/src/db/model.rs#L2136)! But they're not checking the random bits, and now that I look again, could be a bit stronger. For example, checking that the subnet prefix requested is the one actually returned, making sure the remaining bits are 0, etc. I'm happy to add more, including one that explicitly sets the seed to verify the random bits.

Reply:
Added an if-cfg directive to set the seed in a test environment. Please make sure I didn't screw that configuration part up, since explicitly seeding the RNG in a _non-test_ environment would be A Bad Thing.